src/twisted/mail/test/test_mailmail.py

```diff
@@ -73,3 +109,271 @@ def test_copyAliasedSender(self):
+        stdin = NativeStringIO('\n')
+        self.patch(sys, 'argv', argv)
+        self.patch(sys, 'stdin', stdin)
+        mailmail.run()
+        self.assertEqual(self.options.sender, mailmail.getlogin())
+        self.assertEqual(self.options.to, ["invaliduser2@example.com"])
+        # We should have printErrors set because we specified "-oep"
+        self.assertTrue(self.options.printErrors)
+        # We should not have any failures.
+        self.assertIsNone(mailmail.failed)
+
+    if platformType == "win32":
+        test_run.skip = ("mailmail.run() does not work on win32 due to "
+                         "lack of support for getuid()")
+
+
+    def test_readInvalidConfig(self):
+        """
+        Read an invalid configuration file.  Error messages
+        for illegal UID value, illegal GID value, and illegal
+        identity entry will be sent to stderr.
+        """
+        stdin = NativeStringIO('\n')
+        self.patch(sys, 'stdin', stdin)
+
+        filename = self.mktemp()
+        myUid = os.getuid()
+        myGid = os.getgid()
+
+        with open(filename, "w") as f:
+            # Create a config file with some invalid values
+            f.write("[useraccess]\n"
+                    "allow=invaliduser2,invaliduser1\n"
+                    "deny=invaliduser3,invaliduser4,{}\n"
+                    "order=allow,deny\n"
+                    "[groupaccess]\n"
+                    "allow=invalidgid1,invalidgid2\n"
+                    "deny=invalidgid1,invalidgid2,{}\n"
+                    "order=deny,allow\n"
+                    "[identity]\n"
+                    "localhost=funny\n"
+                    "[addresses]\n"
+                    "smarthost=localhost\n"
+                    "default_domain=example.com\n".format(myUid, myGid))
+
+        # The mailmail script looks in
+        # the twisted.mail.scripts.GLOBAL_CFG variable
+        # and then the twisted.mail.scripts.LOCAL_CFG
+        # variable for the path to it's  config file.
+        #
+        # Override twisted.mail.scripts.LOCAL_CFG with the file we just
+        # created.
+        self.patch(mailmail, "LOCAL_CFG", filename)
+
+        argv = ("test_mailmail.py", "invaliduser2@example.com", "-oep")
+        self.patch(sys, 'argv', argv)
+        mailmail.run()
+        self.assertRegex(self.out.getvalue(),
+                         "Illegal UID in \\[useraccess\\] section: "
+                         "invaliduser1")
+        self.assertRegex(self.out.getvalue(),
+                         "Illegal GID in \\[groupaccess\\] section: "
+                         "invalidgid1")
+        self.assertRegex(self.out.getvalue(),
+                         'Illegal entry in \\[identity\\] section: funny')
+
+    if platformType == "win32":
+        test_readInvalidConfig.skip = ("mailmail.run() does not work on win32"
+                                       " due to lack of support for getuid()")
+
+
+    def _loadConfig(self, config):
```

Comment:
may rename this to `getConfigurationFromText` ... or something like that

No need to make it private as this is testing code.

For code which should be reused by multiple tests is better to have it defined as public.

We can keep private methods only for the code which is specific for a single test and should not be reused anywhere else.